packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts

```diff
@@ -0,0 +1,106 @@
+ */
+import { ElementHandle, Frame, Page } from 'playwright';
+import { BaseContainer } from '../base-container';
+
+/**
+ * Represents an instance of the WPCOM's Gutenberg editor page.
+ *
+ * @augments {BaseContainer}
+ */
+export class GutenbergEditorPage extends BaseContainer {
+	frame!: Frame;
+
+	// Page and Frame related selectors.
+	editorSelector = 'div.main.main-column.calypsoify.is-iframe > iframe';
+
+	// Editor element selectors.
+	titleSelector = `.editor-post-title__input`;
+	appenderSelector = '.block-editor-default-block-appender';
+	paragraphSelector = 'p.block-editor-rich-text__editable:first-of-type';
+
+	// Publish flow selectors.
+	firstPublishButtonSelector = '.editor-post-publish-panel__toggle';
+	secondPublishButtonSelector =
+		'.editor-post-publish-panel__header-publish-button button.editor-post-publish-button';
+	snackBarNoticeSelector = '.components-snackbar';
+	snackBarNoticeLinkSelector = '.components-snackbar__content a';
+
+	/**
+	 * Constructs an instance of this object.
+	 *
+	 * @param {Page} page The page where actions take place.
+	 */
+	constructor( page: Page ) {
+		super( page, 'div.main.main-column.calypsoify.is-iframe > iframe' );
+	}
+
+	/**
+	 * Overrides the function of same name defined in the base class.
+	 * This ensures the iframe containing the editor is is fully loaded prior to
+	 * continuing with the test.
+	 *
+	 * @returns {Promise<void>} No return value.
+	 */
+	async _postInit(): Promise< void > {
+		const handle = ( await this.page.$( this.editorSelector ) ) as ElementHandle;
+		this.frame = ( await handle.contentFrame() ) as Frame;
+	}
+
+	/**
+	 * Enters text into the title block.
+	 *
+	 * @param {string} title Text to be used as the title.
+	 * @returns {Promise<void>} No return value.
+	 */
+	async enterTitle( title: string ): Promise< void > {
+		await this.frame.click( this.titleSelector );
+		await this.frame.fill( this.titleSelector, title );
+	}
+
+	/**
+	 * Enters text into the body.
+	 *
+	 * @param {string} text Text to be entered into the body.
+	 * @returns {Promise<void>} No return value.
+	 */
+	async enterText( text: string ): Promise< void > {
+		await this.frame.click( this.appenderSelector );
+		await this.frame.fill( this.paragraphSelector, text );
+	}
+
+	/**
+	 * Publishes the post or page.
+	 *
+	 * @param {boolean} visit Whether to then visit the page.
+	 * @returns {Promise<void} No return value.
+	 */
+	async publish( visit = false ): Promise< void > {
+		await this.frame.click( this.firstPublishButtonSelector );
+		await this.frame.click( this.secondPublishButtonSelector );
+
+		await this.frame.waitForSelector( this.snackBarNoticeSelector );
+
+		if ( visit ) {
+			await this._visitPublishedPost();
+		}
+	}
+
+	/**
+	 * Visits the published post or page.
+	 *
+	 * @returns {Promise<void>} No return value.
+	 */
+	async _visitPublishedPost(): Promise< void > {
+		await Promise.all( [
+			this.frame.click( this.snackBarNoticeLinkSelector ),
```

Comment:
Just wanted to point out here, that the snackbar link was an unreliable (very flaky) way of getting to the published post with Selenium. It was caused by the fact that the snackbar is an animated element and is visible only for a few seconds. In [one of the recent refactors ](https://github.com/Automattic/wp-calypso/pull/52089/commits/9388bd18dae42342ee0b72e07584226c75be1c30#diff-b7f8735b885b065dc18c161fbf8de3c5bfecafd5fd04319842a8eea5f5c32180R74) I replaced the snackbar click with the `View Post` button from the post-publish menu, which fixed the flakiness: 


<p align="center">
<img src='https://user-images.githubusercontent.com/1451471/119643150-b54c4100-be1b-11eb-9831-83e60083cbcb.png' width="240" />
</p>

Another thing here is that it might not serve well as an independent `_visitPublishedPost` method. It's highly dependant on the `publish` method and can be called only at a very specific moment.

Reply:
Aye, I was attempting to stay true to the original Selenium way of doing things to minimize confusion, but it seems that consideration isn't really necessary. 

It should be trivial to rewrite the visit post function to use the `View Post` button that appears in the sidebar. I will likely rename the function to something more constrained eg. `visitEntryFromSidebar` - note the removal of the word 'post' since the DOM uses the term 'entry' as a generic, and the addition of the word 'sidebar' specifically stating this method goes through the sidebar.

Reply:
Replaced the use of `snackbar` with what I call `postPublishPane`.